src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -461,8 +488,10 @@ namespace cryptonote
   //---------------------------------------------------------------
   bool construct_tx(const account_keys& sender_account_keys, const std::vector<tx_source_entry>& sources, const std::vector<tx_destination_entry>& destinations, std::vector<uint8_t> extra, transaction& tx, uint64_t unlock_time)
   {
+     std::unordered_map<crypto::public_key, cryptonote::subaddress_index> subaddresses;
+     subaddresses[sender_account_keys.m_account_address.m_spend_public_key] = {0,0};
```

Comment:
Yes, this is necessary in order to recognize any incoming transfers including those to the original main address, because instead of doing
```
memcmp(P, Hs(aR)G+B)
```
we now do
```
hashtable.find(P-Hs(aR)G)
```
to check if the output belongs to the wallet or not. Thus, the original spend pub key `B` must also be stored in the hashtable to correctly recognize the incoming transfer. See `is_out_to_acc_precomp()` in cryptonote_format_utils.cpp.

Regarding the ordering of the arguments, I did this way since the subaddress hashtable has a similar role as the sender account keys (in that it's necessary for deriving output secret keys) while the `is_subaddress` param provides some additional information about the destination (like the payment ID in extra). I can of course change it if you disagree, though.